### ISSUE TITLE
Enhancement: [require-await] support explicit resource management

### ISSUE THREAD

**Issue by bradzacher** [typescript-eslint/typescript-eslint#11425]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/require-await/

### Description

From the ESLint v9.32.0 release notes:

> [require-await](https://eslint.org/docs/latest/rules/require-await) and [no-await-in-loop](https://eslint.org/docs/latest/rules/no-await-in-loop): Both rules now recognize await using as an await expression for their checks and reporting.

### Fail

```TypeScript
async function foo() { }
```

### Pass

```TypeScript
async function foo() {
  await using foo = disposable();
}
```

**Comment by ahmedsalman74** [comment]
I’d like to work on this issue.
I will start investigating and submit a pull request once I have a solution.
Please assign this issue to me if possible.

**Comment by bradzacher** [comment]
Turns out we do have support -- it was added in https://github.com/typescript-eslint/typescript-eslint/pull/7866.
Duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/7865
